backend/de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/everhour/user/UserImporterService.java

```diff
@@ -0,0 +1,239 @@
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.serviceprovider.everhour.user;
+
+import com.google.common.collect.ImmutableMap;
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.service.IBPartnerBL;
+import de.metas.externalreference.ExternalId;
+import de.metas.externalreference.ExternalReference;
+import de.metas.externalreference.ExternalReferenceQuery;
+import de.metas.externalreference.ExternalReferenceRepository;
+import de.metas.externalreference.ExternalUserReferenceType;
+import de.metas.i18n.Language;
+import de.metas.issue.tracking.everhour.api.EverhourClient;
+import de.metas.issue.tracking.everhour.api.model.User;
+import de.metas.organization.OrgId;
+import de.metas.serviceprovider.external.ExternalSystem;
+import de.metas.user.UserId;
+import de.metas.user.UserQuery;
+import de.metas.user.UserRepository;
+import de.metas.util.Services;
+import de.metas.util.collections.CollectionUtils;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.service.ISysConfigBL;
+import org.compiere.util.Env;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static de.metas.serviceprovider.everhour.EverhourImportConstants.EverhourImporterSysConfig.ACCESS_TOKEN;
+import static de.metas.serviceprovider.everhour.EverhourImportConstants.EverhourImporterSysConfig.BPARTNER_USER_IMPORT;
+
+@Service
+public class UserImporterService
+{
+	private final IBPartnerBL bPartnerBL = Services.get(IBPartnerBL.class);
+	private final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+
+	private final ExternalReferenceRepository externalReferenceRepository;
+	private final UserRepository userRepository;
+	private final EverhourClient everhourClient;
+
+	public UserImporterService(
+			@NonNull final ExternalReferenceRepository externalReferenceRepository,
+			@NonNull final UserRepository userRepository,
+			@NonNull final EverhourClient everhourClient)
+	{
+		this.externalReferenceRepository = externalReferenceRepository;
+		this.userRepository = userRepository;
+		this.everhourClient = everhourClient;
+	}
+
+	@NonNull
+	public UserId resolveUser(@NonNull final OrgId orgId, @NonNull final Integer everhourUserId)
```

Comment:
suggestion: instead of `Integer everhourUserId` i would introduce EverhourUserId which basically wraps an `int`.
Then I would use it all over the place.
This way, it would be clear for the reader and avoid mistakes.

(at least that's what the priest of the DDD church would say :) )

Reply:
changed accordingly